src/main/java/org/broadinstitute/hellbender/utils/mcmc/ParameterizedState.java

```diff
@@ -1,36 +1,104 @@
 package org.broadinstitute.hellbender.utils.mcmc;
 
-import java.util.List;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.util.*;
 
 /**
- * Basic implementation of {@link AbstractParameterizedState} with no convenience fields or getters.  See
- * example of use in GibbsSamplerSingleGaussianUnitTest.
+ * Represents a mapped collection of {@link Parameter} objects, i.e., named, ordered, enumerated keys associated with
+ * values of mixed type via a key -> key, value map.
+ * See GibbsSamplerSingleGaussianUnitTest and GibbsSamplerCopyRatioUnitTest for examples of use.
  *
  * @author Samuel Lee &lt;slee@broadinstitute.org&gt;
  */
-public final class ParameterizedState extends AbstractParameterizedState {
+public class ParameterizedState<T extends Enum<T> & ParameterEnum> {
+    private final LinkedHashMap<T, Parameter<T, ?>> parameterMap;     //must be a Map that gives a sorted keySet
```

Comment:
This comment is confusing -- a `LinkedHashMap` "sorts" its keys by insertion order, but not via a `Comparator`, as a `SortedMap` does.